pandas/tseries/resample.py

```diff
@@ -444,6 +445,17 @@ def _wrap_result(self, result):
 
         return result
 
+    def _assure_downsample(self, _method):
+        """Assure that downsample is done with the correct type"""
+        result = self._downsample(_method)
+
+        # Make consistent type of result. GH14962
+        if not len(self.ax) and isinstance(result, ABCDataFrame) and \
+                isinstance(result.index, (DatetimeIndex, TimedeltaIndex)):
```

Comment:
you can do this logic in ``_wrap_results``, but I again don't think this is ncessary, simply define the ``size`` method as it 'seems' special and needs extra handling.